database/email.go

```diff
@@ -0,0 +1,128 @@
+package database
+
+import (
+	"context"
+	"time"
+
+	"gitlab.com/NebulousLabs/errors"
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/bson/primitive"
+	"go.mongodb.org/mongo-driver/mongo/options"
+)
+
+const (
+	// maxAttemptsToSend defines the maximum number of attempts we will make to
+	// send a given email message before giving up.
+	maxAttemptsToSend = 3
+)
+
+type (
+	// EmailMessage represents an email message waiting to be sent
+	EmailMessage struct {
+		ID                   primitive.ObjectID `bson:"_id,omitempty"`
```

Comment:
why are some fields `omitempty`?

Reply:
If you don't set the `_id` filed to `omitempty` you need to generate the ID yourself, otherwise the record gets inserted in the DB with a `00000000000000` ID which is not what we want. The same is true for `sent_at` where we won't be able to check if it's `nil` in the query but we'll need to compare it to a specific zero value.